src/color.rs

```diff
@@ -160,3 +165,67 @@ impl<C: Color> fmt::Display for Bg<C> {
         self.0.write_bg(f)
     }
 }
+
+/// Types that allow detection of the colors they support
+pub trait DetectColors {
```

Comment:
Well given #88 and the possibility of having other `Write`able interfaces that are raw, I thought a Trait made more sense.

It's also very color related so it felt more correct to have it as an optional thing you can include from the `color` module.